src/build_loss_objective.jl

```diff
@@ -25,10 +25,16 @@ function build_loss_objective(prob::DEProblem,alg,loss,regularization=nothing;mp
       sol = solve(tmp_prob,alg;kwargs...)
     end
 
-    if regularization == nothing
+    if regularization == nothing && prior == nothing
       loss_val = loss(sol)
-    else
+    elseif prior == nothing
       loss_val = loss(sol) + regularization(p)
```

Comment:
I think just adding to a loss val would really clean up this code

Reply:
Didn't quite understand what you mean here?